### PR TITLE
Vendor service-catalog and add to origin.spec

### PR DESCRIPTION
**Pull request by jpeeler**

### CODE REVIEW

File: origin.spec [hunk 1]
```diff
@@ -231,6 +238,30 @@ of docker.  Exclude those versions of docker.
 OS_ONLY_BUILD_PLATFORMS="${BUILD_PLATFORM}" %{os_git_vars} hack/build-cross.sh
 %endif
 
+# Build service-catalog
```
Comment (stevekuznetsov) [inline origin.spec]: Should be able to squash this into the above `if/elif/else` tree:

```spec
%build
%if 0%{make_redistributable}
# Create Binaries for all supported arches
%{os_git_vars} hack/build-cross.sh
%{os_git_vars} cmd/service-catalog/go/src/github.com/kubernetes-incubator/service-catalog/hack/build-cross.sh 
%else
# ...
```

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test] https://github.com/openshift/origin/issues/13980

**Comment by jpeeler** [comment]
[test] #13980

**Comment by aveshagarwal** [comment]
@jpeeler Have you updated origin/hack/build-images.sh to include service catalog image so that the SC image is also built as part of origin/openshift build pipeline?

**Comment by aveshagarwal** [comment]
@jpeeler uodate hack/verify-govet.sh as discussed with eparis if you think thats ok. I update mine.

**Comment by aveshagarwal** [comment]
I am assuming you are already going to update this PR for one rpm/one container.

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by jpeeler** [comment]
[test]

**Comment by eparis** [comment]
[merge] you fool! merge I say!

**Comment by jpeeler** [comment]
[test]

**Comment by eparis** [comment]
[merge] after rebase

**Comment by jpeeler** [comment]
[test]

**Comment by eparis** [comment]
and since you edit'd it we need another [merge]

**Comment by jpeeler** [comment]
[test]

**Comment by eparis** [comment]
[test] because of flaky stuff?

**Comment by eparis** [comment]
[merge]

**Comment by aveshagarwal** [comment]
@jpeeler I am confused why it seems that you removed hellp-openshift image and added s-c image hack/push-release.sh? sorry if I missed something.

**Comment by eparis** [comment]
[merge] (but with 24 in the queue, it isn't going to matter any time soon)

File: origin.spec [hunk 2]
```diff
@@ -301,6 +309,11 @@ install -p -m 755 _output/local/bin/${PLATFORM}/hyperkube %{buildroot}%{_bindir}
 install -p -m 755 cmd/cluster-capacity/go/src/github.com/kubernetes-incubator/cluster-capacity/_output/local/bin/${PLATFORM}/hypercc %{buildroot}%{_bindir}/
 ln -s hypercc %{buildroot}%{_bindir}/cluster-capacity
 
+# Install service-catalog
+install -p -m 755 cmd/service-catalog/go/src/github.com/kubernetes-incubator/service-catalog/_output/local/bin/${PLATFORM}/apiserver %{buildroot}%{_bindir}/
+install -p -m 755 cmd/service-catalog/go/src/github.com/kubernetes-incubator/service-catalog/_output/local/bin/${PLATFORM}/controller-manager %{buildroot}%{_bindir}/
+install -p -m 755 cmd/service-catalog/go/src/github.com/kubernetes-incubator/service-catalog/_output/local/bin/${PLATFORM}/user-broker %{buildroot}%{_bindir}/
```
Comment (derekwaynecarr) [inline origin.spec]: @pmorie - why are we shipping this?  we should remove the user-broker

File: origin.spec [hunk 3]
```diff
@@ -549,6 +562,11 @@ if [ -d %{kube_plugin_path} ]; then
   rmdir %{kube_plugin_path}
 fi
 
+%files service-catalog
+%{_bindir}/apiserver
+%{_bindir}/controller-manager
+%{_bindir}/user-broker
```
Comment (derekwaynecarr) [inline origin.spec]: same here...

**Comment by eparis** [comment]
wait the [test] failed because it logged in too quickly?

**Comment by stevekuznetsov** [comment]
lol

**Pull request merged**